### PR TITLE
User/project population script

### PR DESCRIPTION
**Pull request by srm912**
This is an initial version of the data population script. Currently,
this script takes as input
1. The number of users to be created
2. The number of projects to be created

Currently, all the created users have projectAdmin as their assigned
roles. This will change the in the forthcoming commits.
This script is the first part of a verification test suite which can be
leveraged to check the consistency of data with various data backends.

### CODE REVIEW

File: src/dataservice/data.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        "time"
+
+        "github.com/vmware/harbor/src/common/models"
+        "github.com/manveru/faker"
+)
+
+// Harbor deployment URL
+var baseURL string
+
+// This instantiates a Faker module used to generate randomized data
```
Comment (vuil) [inline src/dataservice/data.go]: Does Faker generate random data deterministically or can it be made to do so? Having that behavior would likely make the validations we have to do in the tests employing this tool easier to write.

Comment (srm912) [inline src/dataservice/data.go]: AFAIK there are no provisions to control the data generated. It has limited provisions to tweak the data generated.
For now, we use Faker to create the Project name and the user details i.e. name, username etc

File: src/dataservice/data.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    r, _ := http.NewRequest("POST", urlStr, bytes.NewBufferString(data.Encode()))
+    r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+    r.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+
+    resp, _ := client.Do(r)
+    return resp.Cookies()[0].Value
+}
+
+func getJSON(baseURL string, endpoint string, cookie http.Cookie) *http.Response {
+
```
Comment (vuil) [inline src/dataservice/data.go]: nit: There are several places with the leading blank line in functions, can we remove them?

File: src/dataservice/data.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    if err != nil {
+       panic(err.Error())
+    }
+    defer resp.Body.Close()
+    var projects = new([]models.Project)
+    json.Unmarshal([]byte(body), &projects)
+    return *projects
+}
+
+// Always returns 404, tried with POSTMAN client as well
```
Comment (vuil) [inline src/dataservice/data.go]: Is this not longer valid? Or are we saying this doesn't quite work?

File: src/dataservice/data.go (hunk 2 above)
Comment (srm912) [inline src/dataservice/data.go]: Done

File: tests/loadgen/data.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    os.Exit(1)
+}
+
+func verifyProjectsForUser(user models.User, projects []models.Project, dataGen DataGen) (bool, string) {
+    var projectImages = dataGen.GetProjects(user.Username)
+    var isEqual = true
+    var msg = ""
+    for i:=0; i<len(projectImages) && isEqual; i++ {
+        projReq := projectImages[i].ProjectReq
+        //TODO: Remove the comapred project from the projects[]
```
Comment (vuil) [inline tests/loadgen/data.go]: typo

File: tests/loadgen/datagen.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
+
+package main
+
+import (
+        //"fmt"
```
Comment (vuil) [inline tests/loadgen/datagen.go]: remove

File: tests/loadgen/datagen.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        }
+    }
+    return projectImages
+}
+
+func (d DataGen) GetProjects(username string) []ProjectImage {
+    return d.UserProjectMapping[username]
+}
+
+// func (d DataGen) GetProject(projectName string) (ProjectReq, models.User) {
```
Comment (vuil) [inline tests/loadgen/datagen.go]: do you mean to remove this?

File: tests/loadgen/datagen.go [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+
+func _randInt(min, max int) int {
+    return min + rand.Intn(max-min)
+}
+
+func _genProjectName() string {
+    faker := getInstance()
+    //var name = faker.Words(1, false)[0]
```
Comment (vuil) [inline tests/loadgen/datagen.go]: do yo mean to remove this?

**Comment by srm912** [comment]
Currently the script creates the users and then creates projects using these users.
The owner of each project is used to push images to the repo.

The script has no checks, but Harbor has checks against it.

File: tests/loadgen/datagen.go (hunk 7 above)
Comment (srm912) [inline tests/loadgen/datagen.go]: Done

File: tests/loadgen/datagen.go (hunk 6 above)
Comment (srm912) [inline tests/loadgen/datagen.go]: Done

File: tests/loadgen/datagen.go (hunk 5 above)
Comment (srm912) [inline tests/loadgen/datagen.go]: Done

File: tests/loadgen/data.go (hunk 4 above)
Comment (srm912) [inline tests/loadgen/data.go]: Done

**Pull request merged**